app/services/migration/migrators/user.rb

```diff
@@ -54,6 +54,72 @@ def call
 
   private
 
+    def find_or_initialize_user(ecf_user)
```

Comment:
This is a bit of a monster function; can we extract several methods to make it easier to follow?

Reply:
I'm not sure, i think the readability is important here as we are dealing with various scenarios, just running through step by step.

Reply:
I think I agree around breaking this up, but I think we're probably going to move this out into the deduper/service anyway in your current ticket Mook, so maybe we break up there rather than here?

Reply:
I think it would be much easier to digest/maintain/extend if it was refactored into multiple smaller, well named functions

Reply:
agreed, this will be addressed shortly hopefully in the followup pr :)